cmd/manager/main_test.go

```diff
@@ -23,3 +33,75 @@ func TestCheckIfRequiredFilesExist(t *testing.T) {
 		"Expected error message is absent")
 
 }
+
+// TestIsValidKubernetesVersion tests if isValidKubernetesVersion function throws error if K8s version is less than v1.18.0
+func TestIsValidKubernetesVersion(t *testing.T) {
+	fakeConfigClient := fakeconfigclient.NewSimpleClientset()
+	// specify version to be tested
+	fakeConfigClient.Discovery().(*fakediscovery.FakeDiscovery).FakedServerVersion = &version.Info{
+		GitVersion: "v1.17.0",
+	}
+	err := isValidKubernetesVersion(fakeConfigClient)
+	require.Error(t, err, "Function isValidKubernetesVersion did not throw an error when it was expected to")
+	assert.Contains(t, err.Error(), "Unsupported server version: v1.17.0. Supported version is v1.18.0.")
+}
+
+// TestNetworkConfigurationFactory tests if TestNetworkConfigurationFactory function throws appropriate errors
+func TestNetworkConfigurationFactory(t *testing.T) {
+	var tests = []struct {
+		name         string
+		networkType  string
+		errorMessage string
+	}{
+		{"invalid network type", "OpenShiftSDN", "OpenShiftSDN : network type not supported"},
+		{"valid network type", "OVNKubernetes", ""},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeConfigClient, fakeOperatorClient := createFakeClients(tt.networkType)
+
+			_, err := clusternetwork.NetworkConfigurationFactory(fakeConfigClient, fakeOperatorClient)
+			if tt.errorMessage == "" {
+				require.Nil(t, err, "Successful check for valid network type")
+			} else {
+				require.Error(t, err, "Function networkConfigurationFactory did not throw an error "+
+					"when it was expected to")
+				assert.Contains(t, err.Error(), tt.errorMessage)
+			}
+		})
+	}
+}
+
+// TestValidate tests if validate() method throws error when network is OVNKubernetes but hybrid overlay is not enabled
+func TestValidate(t *testing.T) {
```

Comment:
This test is named as if it will be testing the `Validate()` interface, but it only tests the `OVNKubernetes` implementation of the interface. You can either rename the test to something more specific, or you can generalize this test to allow for future `NetworkConfiguration` implementations to be plugged in

Reply:
You have not addressed this

Reply:
https://github.com/openshift/windows-machine-config-operator/pull/37#issuecomment-614796424

Please rename it to `TestNetworkConfigurationValidate` as `TestValidate` is too generic.

Reply:
Shouldn't this unit test be in the `clusternetwork` package?

Reply:
Yes, will move `TestValidate` to `clusternetwork`